versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5771 @@
+  >>>
+
+  output {
+    String wizard_name = "${w.name} Potter"
+    Int age_in_muggle_years = w.age * 2
+  }
+    
+  runtime {
+    container: "my_image:latest"
+  }
+}
+
+workflow my_workflow {
+  Wizard harry = Wizard { name: "Harry", age: 11 }
+  
+  call myTask { input: a = harry }
+}
+```
+
+Access to elements of compound members can be chained into a single expression. For example:
+
+```wdl
+struct Experiment {
+  Array[File] experimentFiles
+  Map[String, String] experimentData
+}
+```
+
+**Example 1:** Accessing the nth element of experimentFiles and any element in experimentData:
+
+```wdl
+workflow workflow_a {
+  input {
+    Experiment myExperiment
+  }
+  File firstFile = myExperiment.experimentFiles[0]
+  String experimentName = myExperiment.experimentData["name"]
+}
+```
+
+**Example 2:** The struct is an item in an Array:
+
+```wdl
+workflow workflow_a {
+  input {
+    Array[Experiment] myExperiments
+  }
+
+  File firstFileFromFirstExperiment = myExperiments[0].experimentFiles[0]
+  File experimentNameFromFirstExperiment = myExperiments[0].experimentData["name"]
+  ....
+}
+```
+
+# Standard Library
+
+The following functions are available to be called in WDL expressions. The signature of each function is given as `R func_name(T1, T2, ...)`, where `R` is the return type and `T1`, `T2`, ... are the parameter types. All function parameters must be specified in order, and all function parameters are required, with the exception that the last parameter of some functions is optional (denoted by the type in brackets `[]`).
+
+A function is called using the following syntax: `R' val = func_name(arg1, arg2, ...)`, where `R'` is a type that is coercible from `R`, and `arg1`, `arg2`, ... are expressions whose types are coercible to `T1`, `T2`, ...
+
+Some functions accept arguments of multiple different types, denoted as a list of types separated by `|`.
+
+Some functions are polymorphic, which means that they are actually multiple functions with the same name but different signatures. Such functions are defined with generic types (e.g. `X`, `Y`) instead of concrete types (e.g. `File` or `String`), and the bounds of each type parameter is specified in the function description.
+
+Functions that are new in this version of the specification are denoted by ✨, and deprecated functions are denoted by 🗑.
+
+## Int floor(Float), Int ceil(Float) and Int round(Float)
+
+These functions convert a `Float` to an `Int` using different rounding methods:
+
+- `floor`: Rounds **down** to the next lower integer.
+- `ceil`: Rounds **up** to the next higher integer.
+- `round`: Rounds to the nearest integer based on standard rounding rules ("round half up").
+
+**Parameters**:
+
+1. `Float`: the number to round.
+
+**Returns**: An integer.
+
+**Example**
+
+```wdl
+floor(1.0) == 1
+floor(1.9) == 1
+ceil(2.0) == 2
+ceil(2.1) == 3
+round(1.49) == 1
+round(1.50) == 2
+```
+
+## ✨ Int min(Int, Int), Float min(Int|Float, Int|Float)
```

Comment:
`Int min(Int, Int), Float min(Float, Int), Float min(Int, Float), Float min(Float, Float)` is more correct. The current notation leaves room for `Float min(Int, Int)` which is incorrect. In this case I think spelling it out adds to clarity.

Reply:
You're right, I was just trying to shorten it, but I'll spell it out.